server/lib/report_server_web/components/custom_components.ex

```diff
@@ -163,4 +163,94 @@ defmodule ReportServerWeb.CustomComponents do
     """
   end
 
+  attr :previous, :any, required: true
+  attr :current, :string, required: true
+  def breadcrumbs(assigns) do
+    ~H"""
+    <span :for={{title, path} <- @previous}>
+      <.link navigate={path} class="hover:underline"><%= title %></.link>
+      <span>›</span>
+    </span>
+    <%= @current %>
+    """
+  end
+
+  attr :report_run, :any, required: true
+  def report_filter_values(assigns) do
+    ~H"""
+    <div class="table">
+      <div class="table-row" :for={filter <- Enum.reverse(@report_run.report_filter.filters)}>
+        <div class="table-cell capitalize font-bold"><%= filter %>s</div>
+        <div class="table-cell pl-3"><%= Enum.join(Map.values(@report_run.report_filter_values[filter] || %{}), ", ") %></div>
+      </div>
+    </div>
+    """
+  end
+
+  attr :timestamp, :string, required: true
+  def relative_time(assigns) do
+    now = DateTime.utc_now() |> DateTime.to_unix()
+    diff = now - DateTime.to_unix(assigns.timestamp)
+
+    relative_time_result = cond do
+      diff < 60 ->
+        "Just now"
+
+      diff < 3600 ->
+        minutes = div(diff, 60)
+        "#{minutes} #{pluralize(minutes, "minute")} ago"
+
+      diff < 86_400 ->
+        hours = div(diff, 3600)
+        "#{hours} #{pluralize(hours, "hour")} ago"
+
+      true ->
+        days = div(diff, 86_400)
+        "#{days} #{pluralize(days, "day")} ago"
+    end
+
+    assigns = assign(assigns, :relative_time_result, relative_time_result)
+
+    ~H"<%= @relative_time_result %>"
+  end
+
+  attr :report_runs, :any, required: true
+  attr :include_report_titles, :boolean, default: false
+  attr :include_user, :boolean, default: false
+  def report_runs(assigns) do
+    report_titles = Enum.reduce(assigns.report_runs, %{}, fn %{report_slug: report_slug}, acc ->
+      report = ReportServer.Reports.Tree.find_report(report_slug)
+      title = if report, do: report.title, else: "Unknown report: #{report_slug}"
+      Map.put(acc, report_slug, title)
+    end)
+    assigns = assign(assigns, :report_titles, report_titles)
+
+    ~H"""
+      <table class="w-full border-collapse bg-white text-sm">
+        <thead class="bg-gray-100 text-left leading-6 text-zinc-500">
+          <tr>
+            <th class="p-2 font-normal border-b">Run</th>
+            <th class="p-2 font-normal border-b" :if={@include_user}>User</th>
+            <th class="p-2 font-normal border-b" :if={@include_report_titles}>Report</th>
+            <th class="p-2 font-normal border-b">Filters</th>
+            <th class="p-2 font-normal border-b">Ran</th>
+          </tr>
+        </thead>
+        <tbody>
+          <tr :for={report_run <- @report_runs} class="group hover:bg-zinc-200 even:bg-gray-50">
+            <td class="p-2 font-normal border-b align-top"><.link class="underline" href={"/new-reports/runs/#{report_run.id}"}><%= report_run.id %></.link></td>
```

Comment:
The ID number is kind of meaningless to the end user. OK to include it for test/debug purposes, but clicking it to re-run the report is not very obvious.  There should probably be a "re-run" button instead.

Reply:
I can see Trudi memorizing the report run ids she wants to run during a class test.  We could also add a view buttom on the far right.